skyportal/handlers/api/source.py

```diff
@@ -52,54 +57,72 @@ def get(self, source_id=None):
-                         .joinedload(Photometry.instrument)
-                         .joinedload(Instrument.telescope)])
+                register_source_view(
+                    source_id=source_id,
+                    username_or_token_id=self.current_user.id,
+                    is_token=True,
+                )
+            info["sources"] = Source.get_if_owned_by(
+                source_id,
+                self.current_user,
+                options=[
+                    joinedload(Source.comments),
+                    joinedload(Source.thumbnails)
+                    .joinedload(Thumbnail.photometry)
+                    .joinedload(Photometry.instrument)
+                    .joinedload(Instrument.telescope),
+                ],
+            )
         elif page_number:
             try:
                 page = int(page_number)
             except ValueError:
                 return self.error("Invalid page number value.")
-            q = Source.query.filter(Source.id.in_(DBSession.query(
-                GroupSource.source_id).filter(GroupSource.group_id.in_(
-                    [g.id for g in self.current_user.groups]))))
+            q = Source.query.filter(
+                Source.id.in_(
+                    DBSession.query(GroupSource.source_id).filter(
+                        GroupSource.group_id.in_(
+                            [g.id for g in self.current_user.groups]
+                        )
+                    )
+                )
+            )
             if sourceID:
                 q = q.filter(Source.id.contains(sourceID.strip()))
             if any([ra, dec, radius]):
                 if not all([ra, dec, radius]):
-                    return self.error("If any of 'ra', 'dec' or 'radius' are "
-                                      "provided, all three are required.")
+                    return self.error(
+                        "If any of 'ra', 'dec' or 'radius' are "
+                        "provided, all three are required."
+                    )
                 try:
                     ra = float(ra)
                     dec = float(dec)
                     radius = float(radius)
                 except ValueError:
-                    return self.error("Invalid values for ra, dec or radius - could not convert to float")
-                q = q.filter(Source.ra <= ra + radius)\
-                     .filter(Source.ra >= ra - radius)\
-                     .filter(Source.dec <= dec + radius)\
-                     .filter(Source.dec >= dec - radius)
+                    return self.error(
+                        "Invalid values for ra, dec or radius - could not convert to float"
+                    )
+                q = (
+                    q.filter(Source.ra <= ra + radius)
+                    .filter(Source.ra >= ra - radius)
+                    .filter(Source.dec <= dec + radius)
+                    .filter(Source.dec >= dec - radius)
```

Comment:
this is not a true cone search: it will fail near the poles and also near ra 360->0.

Reply:
Yeah, that's just been a placeholder for now. I think there's already an open issue for that, but I'll open one now if not. Since this isn't new in this PR, we won't handle it here (it's only in the diff due to formatting changes).